atc/exec/check_step.go

```diff
@@ -105,59 +112,33 @@ func (step *CheckStep) Run(ctx context.Context, state RunState) error {
 		expires,
 	)
 
-	chosenWorker, err := step.pool.FindOrChooseWorkerForContainer(
+	checkable := step.resourceFactory.NewResource(
+		source,
+		nil,
+		step.plan.FromVersion,
+	)
+
+	result, err := step.workerClient.RunCheckStep(
 		ctx,
 		logger,
 		owner,
 		containerSpec,
 		workerSpec,
 		step.strategy,
-	)
-	if err != nil {
-		logger.Error("failed-to-find-or-choose-worker", err)
-		return err
-	}
 
-	container, err := chosenWorker.FindOrCreateContainer(
-		ctx,
-		logger,
-		step.delegate,
-		owner,
 		step.containerMetadata,
-		containerSpec,
 		resourceTypes,
-	)
-	if err != nil {
-		logger.Error("failed-to-find-or-create-container", err)
-		return err
-	}
-
-	timeout, err := time.ParseDuration(step.plan.Timeout)
-	if err != nil {
-		logger.Error("failed-to-parse-timeout", err)
-		return err
-	}
-
-	deadline, cancel := context.WithTimeout(ctx, timeout)
-	defer cancel()
 
-	processSpec := runtime.ProcessSpec{
-		Path: "/opt/resource/check",
-	}
-
-	checkable := step.resourceFactory.NewResource(source, nil, step.plan.FromVersion)
-	versions, err := checkable.Check(deadline, processSpec, container)
+		timeout,
+		checkable,
+	)
 	if err != nil {
-		if err == context.DeadlineExceeded {
-			return fmt.Errorf("Timed out after %v while checking for new versions", timeout)
-		}
-		return err
+		return fmt.Errorf("run check step: %w", err)
 	}
 
-	err = step.delegate.SaveVersions(versions)
+	err = step.delegate.SaveVersions(result.Versions)
 	if err != nil {
-		logger.Error("failed-to-save-versions", err)
-		return err
+		return fmt.Errorf("save versions: %w", err)
 	}
 
 	step.succeeded = true
```

Comment:
Maybe we could give `CheckResult` an `ExitStatus` and make this similar to:
https://github.com/concourse/concourse/blob/aac7b2f461ed2a3ad6fa8d861d439d3774f9fde8/atc/exec/task_step.go#L248